.circleci/base.yml

```diff
@@ -1,20 +1,37 @@
 version: 2.1
 
 commands:
-  setup:
-    description: "Setup skip toolchain"
+  custom-checkout:
+    description: "Shallow checkout code"
+    parameters:
+      submodules:
+        type: boolean
+        default: false
     steps:
       - run:
-          name: Fetch submodules
+          name: Shallow checkout code
           command: |
-            git submodule update --init --recursive
+            if [ -z "$CIRCLE_BRANCH" ] || [ -n "$CIRCLE_TAG" ]; then echo "TODO: handle no-branch or tag-triggered CI"; exit 1; fi
+            mkdir -p ~/.ssh
+            echo 'github.com ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEAq2A7hRGmdnm9tUDbO9IDSwBK6TbQa+PXYPCPy6rbTrTtw7PHkccKrpp0yVhp5HdEIcKr6pLlVDBfOLX9QUsyCOV0wzfjIJNlGEYsdlLJizHhbn2mUjvSAHQqZETYP81eFzLQNnPHt4EVVUh7VfDESU84KezmD5QlWpXLmvU31/yMf+Se8xhHTvKSCZIFImWwoG6mbUoWf9nzpIoaSjB+weqqUUmpaaasXVal72J+UX2B+2RPW3RcT0eOzQgqlJL3RKrTJvdsjE3JEAvGq3lGHSZXy28G3skua2SmVi/w4yCE6gbODqnTWlg7+wC604ydGXA8VJiS5ap43JXiUFFAaQ==' >> ~/.ssh/known_hosts
+            cat $CHECKOUT_KEY > ~/.ssh/id_rsa
+            chmod 0600 ~/.ssh/id_rsa
+            git config --global url."ssh://git@github.com".insteadOf "https://github.com" || true
```

Comment:
Why is ssh needed here?

Reply:
I assumed it could be a bit faster but probably useless in our case